disnake/automod.py

```diff
@@ -296,7 +302,7 @@ def __init__(
         ...
 
     @overload
-    def __init__(self, *, mention_total_limit: int) -> None:
+    def __init__(self, *, mention_total_limit: int, mention_raid_protection_enabled: bool) -> None:
```

Comment:
`mention_raid_protection_enabled` here should have some sort of default value, probably `False` or `None`?
The runtime default will always be `None`, but the overload should have a default value too.

Not sending it when creating a rule results in the API setting it to `false`; on the other hand, it's set to `true` by default in new guilds.
(apparently it's some sort of builtin rule with a hardcoded ID (`1030554520465440818`) now? it's a bit weird.)

Reply:
> `mention_raid_protection_enabled` here should have some sort of default value, probably `False` or `None`?

Hmm, if we do that, then we should set a default to `mention_total_limit` too.

Reply:
why?
It'd be similar to keyword rules, where `keyword_filter` is required (no default), and `allow_list` is optional (defaulting to `None`).

Reply:
Now that I think about it, you're right, we should set a default value. But should it be true or false? 🤔

Reply:
~~bump~~, I guess it should be false (although it's true when the rule is deleted)

Reply:
> I guess it should be false
yeah, makes sense.

Reply:
https://github.com/DisnakeDev/disnake/pull/898/commits/6a402e35cd0c2abfea50453c7d6e8e788250285c 😁